src/main/java/net/helix/pendulum/controllers/RoundViewModel.java

```diff
@@ -255,7 +255,9 @@ public static int getRoundIndex(TransactionViewModel milestoneTransaction) {
                 }
             } else {
                 Set<Hash> prevMilestones = prevMilestone.getHashes();
-                List<List<Hash>> merkleTree = Merkle.buildMerkleTree(new ArrayList<>(prevMilestones));
+                List<Hash> prevMilestonesList = new ArrayList<>(prevMilestones);
```

Comment:
Let's add trace statements to each `if` or `else` branch so that we can at least figure out how the execution went through looking at the logs